src/lib/layer.js

```diff
@@ -116,37 +116,29 @@ export default class Layer {
     }
   }
 
-  // If state has a model, draw it with supplied uniforms
-  /* eslint-disable max-statements */
-  pick({info, uniforms, pickEnableUniforms, pickDisableUniforms, mode}) {
-    const {gl} = this.context;
-    const {model} = this.state;
+  // called on mouse event.
+  // @returns {bool} handled
+  pick({info, mode}) {
+
+    // TODO - selectedPickingColor should be removed?
+    if (mode === 'hover') {
+      const {color} = info;
+      const selectedPickingColor = new Float32Array(3);
+      selectedPickingColor[0] = color[0];
+      selectedPickingColor[1] = color[1];
+      selectedPickingColor[2] = color[2];
+      this.setUniforms({selectedPickingColor});
```

Comment:
Per discussions with @shaojingli, this should be handled by the JS part of a shader picking shader package. Could change the TODO comment to indicate that.